src/pages/Background/modules/Handler401.js

```diff
@@ -0,0 +1,27 @@
+//This function is to handle expiration of the temporory access token which was obtained through chrome.identity.getAuthToken 
+//the token is only valid for around 45 minutes or so
+//so after expiration, if any API fetch request is made to appScript, it will return a response object with status 401,
+//for all API fetch requests we could handle this unAuthenticated request error by calling this Handler401 function 
+//it will simply regenerate the token value which will be updated in userCreds in local
+//but this time with No consent or login requirement.
+//and we can call the API fetch function again with the updated token value from userCreds stored in local 
+
+function Handler401() {
```

Comment:
What happens when the user does many changes back-to-back when the excel is unauthorised. how are you handling parallel requests failing?

Reply:
UI and sheet/excel are not in sync with each other in real time here.. once the user is done with all the changes then only the the data on local will be loaded on sheet (may be we can provide a commit button for that)